cli/command/stack/deploy_composefile.go

```diff
@@ -320,8 +320,20 @@ func deployServices(
 
 			if resolveImage == resolveImageAlways || (resolveImage == resolveImageChanged && image != service.Spec.Labels[convert.LabelImage]) {
 				updateOpts.QueryRegistry = true
+			} else {
```

Comment:
In this case: `resolveImage == resolveImageNever && image !=  service.Spec.Labels[convert.LabelImage]` we would set the image back to the previous image, which is wrong, because the user requested an update to the image.

I think this needs to be:

```go
switch {
// image should be updated by the server using QueryRegistry
case resolveImage == resolveImageAlways || (resolveImage == resolveImageChanged && image != service.Spec.Labels[convert.LabelImage]):
    updateOpts.QueryRegistry = true
// image hasn't changed, so the serviceSpec needs to be updated to use the existing image that was 
// set by QueryRegistry on the previous deploy. Otherwise this will result in an incorrect service update.
case image == service.Spec.Labels[convert.LabelImage]:
    serviceSpec.TaskTemplate.ContainerSpec.Image = service.Spec.TaskTemplate.ContainerSpec.Image
}
```